### PR TITLE
Add PDF visualize script

### LINKED ISSUE #12: Improve mask visualization
RTBA.

### PR DESCRIPTION
**Pull request by michael153**
This pull request fixes #12

### CODE REVIEW

File: pdf_visualize.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
+"""This module implements a function for effectively visualizing a model's prediction
+versus the ground truth."""
```
Comment (bveeramani) [inline pdf_visualize.py]: Nit: You could be more concise and end the summary at "model's predictions."

File: pdf_visualize.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+def generate_pdf_page(image, mask, ground_truth, classes):
+    """Utility function for visualizing model predictions.
+
+    Arguments:
+        image (tensor): A tensor of shape (C, H, W) representing an image.
+        mask (tensor): A tensor of shape (H, W) representing predicted targets.
+        ground_truth (tensor): A tensor of shape (H, W) representing ground truth targets.
+        classes (list): A list of class names.
+    """
+    # torch.Size([C, H, W]) => torch.Size([H, W, C])
```
Comment (bveeramani) [inline pdf_visualize.py]: Nit: Move this one line down so it's closer to where the actual permutation happens.

File: pdf_visualize.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    num_classes = len(classes)
+    image = image.permute(1, 2, 0)
+
+    fig = plt.figure()
+    fig.add_subplot(141)
```
Comment (bveeramani) [inline pdf_visualize.py]: Nit: What is 141? It seems kind of arbitrary. I would recommend having a named variable (i.e `x_pos = 141` or whatever, I'm unsure what it represents) rather than adding a comment.

File: pdf_visualize.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+    plt.imshow(image)
+    plt.title('Original Image')
+
+    fig.add_subplot(142)
```
Comment (bveeramani) [inline pdf_visualize.py]: Nit: Same as above.

File: pdf_visualize.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    plt.title('Predictions')
+    plt.imshow(mask, vmin=0, vmax=len(classes) - 1, alpha=0.9)
+    plt.axis('off')
+
+    fig.add_subplot(143)
```
Comment (bveeramani) [inline pdf_visualize.py]: Nit: Same as above.

File: pdf_visualize.py [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    patches = [
+        mpatches.Patch(color=colors[i], label=classes[i])
+        for i in range(num_classes)
+    ]
+    plt.legend(handles=patches, bbox_to_anchor=(1.05, 1), loc=2)
+
+    return fig
+
+
+def pdf_visualize(source_name, model_parameter_path, num_entries, pdf_dest_path='output.pdf'):
```
Comment (bveeramani) [inline pdf_visualize.py]: Nit: Rename to `visualize_pdf` so that the function name is a verb.
Nit: Rename `source_name` to `blob_name` or `source_blob_name`. The reason being that source is ambiguous (model source? dataset source?) but `blob_name` is pretty clearly related to GCP storage.

File: pdf_visualize.py [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    """Function for generating PDF compilation of model prediction versus ground truth.
```
Comment (bveeramani) [inline pdf_visualize.py]: Nit: using the phrase "comparing model predictions to ground truth" instead of "model prediction versus ground truth" might make more sense.

File: pdf_visualize.py [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+    Arguments:
+        source_name (string): The source_blob_name for the dataset
```
Comment (bveeramani) [inline pdf_visualize.py]: Nit: This is super minor, but I the convention is (str) instead of (string) since str is the name of the built-in type.

Comment (bveeramani) [inline pdf_visualize.py]: Nit: Also, add periods for consistency with the rest of the documentation in eyecite-ml.

File: pdf_visualize.py [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+        model_parameter_path (string): The path to the model's parameters
+        num_entries (int): The number of entries to visualize
+        pdf_dest_path (string): File path to destination pdf
+
+    Example:
+        pdf_visualize("publaynet-stub.zip", "model.pt", 5)
```
Comment (bveeramani) [inline pdf_visualize.py]: Nit: Prepend ">>>" to line 66 so that it works with the doctest.

File: pdf_visualize.py [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+        figs.append(
+            generate_pdf_page(image, predictions, mask, dataset.CLASSES))
+
+    pdf = matplotlib.backends.backend_pdf.PdfPages(pdf_dest_path)
+    for fig in figs:
+        pdf.savefig(fig)
+
+    pdf.close()
+
+pdf_visualize("publaynet-stub.zip", "model.pt", 5)
```
Comment (bveeramani) [inline pdf_visualize.py]: Enclose this in an `if __name__ == __main__` clause. Without it, we won't be able to import the analysis module without constructing a pdf.

File: pdf_visualize.py [hunk 11]
```diff
@@ -0,0 +1,102 @@
+"""This module implements a function for effectively visualizing a model's prediction."""
+
+import torch
+import matplotlib.pyplot as plt
+import matplotlib.patches as mpatches
+import matplotlib.backends.backend_pdf
+from eyecite import transforms, datasets, models, training
+
+
+#pylint: disable=too-many-locals
```
Comment (bveeramani) [inline pdf_visualize.py]: Nit: I think if you place it here it's disabled for the entire module and not just the function.

File: pdf_visualize.py [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+    """Utility function for visualizing model predictions.
+
+    Arguments:
+        image (tensor): A tensor of shape (C, H, W) representing an image.
+        mask (tensor): A tensor of shape (H, W) representing predicted targets.
+        ground_truth (tensor): A tensor of shape (H, W) representing ground truth targets.
+        classes (list): A list of class names.
+    """
+    num_classes = len(classes)
+    image_pos, pred_pos, ground_pos = 141, 142, 143
```
Comment (bveeramani) [inline pdf_visualize.py]: Nit: It may be more readable if these variables are declared near where they are used (in the `fig.add_subplot(...)`)
